src/dmd/glue.d

```diff
@@ -917,8 +917,15 @@ void FuncDeclaration_toObjFile(FuncDeclaration fd, bool multiobj)
         char[5+4+1] hiddenparam = void;
         __gshared int hiddenparami;    // how many we've generated so far
 
-        sprintf(hiddenparam.ptr,"__HID%d",++hiddenparami);
-        shidden = symbol_name(hiddenparam.ptr,SCparameter,thidden);
+        const(char)* name;
+        if (fd.nrvo_can && fd.nrvo_var)
+            name = fd.nrvo_var.ident.toChars();
+        else
+        {
+            sprintf(hiddenparam.ptr, "__HID%d", ++hiddenparami);
```

Comment:
It would be nice to not have to use C strings. Perhaps you can copy or move this utility function [1].

[1] https://github.com/dlang/dmd/blob/806f105da1d802919f5741ecc1568c9623e5377f/src/dmd/objc_glue.d#L521-L543

Reply:
This use of sprintf was there before and ubiquitous in the dmd source code. I'd rather leave it to some other PR to change that.

Reply:
Right, didn't look close enough on what was removed.